src/chat.rs

```diff
@@ -2804,7 +2806,47 @@ pub(crate) async fn create_send_msg_jobs(context: &Context, msg: &mut Message) -
         return Ok(Vec::new());
     }
 
-    let rendered_msg = match mimefactory.render(context).await {
+    // render message and pre message.
+    // pre message is a small message with metadata
+    // which announces a larger message. Large messages are not downloaded in the background.
+
+    let needs_pre_message = msg.viewtype.has_file()
+        && msg
+            .get_filebytes(context)
+            .await?
+            .context("filebytes not available, even though message has attachment")?
+            > PRE_MESSAGE_ATTACHMENT_SIZE_THRESHOLD;
```

Comment:
Are there plans to also sum the message text size? Text usually compresses better, but we could add e.g. 3/4 of its size.

Reply:
for now this is exclusively about attachments. I think after attachments the next biggest thing may be webxdc updates, before we need to worry about huge texts.